src/storage-types/src/errors.rs

```diff
@@ -375,7 +375,8 @@ impl Display for SourceError {
 #[derive(Arbitrary, Ord, PartialOrd, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash)]
 pub enum SourceErrorDetails {
     Initialization(String),
-    Other(String),
```

Comment:
Is it true that all non-initialization errors are permanent? If we're not sure it seems relatively safe to add a third case.